spa/src/pages/ConfigSteps/index.tsx

```diff
@@ -264,14 +264,14 @@ const ConfigSteps = () => {
 		const handler = async (event: MessageEvent) => {
 			if (event.origin !== originalUrl) return;
 			if (event.data?.type === "oauth-callback" && event.data?.code) {
-				const response = await OAuthManager.finishOAuthFlow(event.data?.code, event.data?.state);
+				const response: boolean | AxiosError = await OAuthManager.finishOAuthFlow(event.data?.code, event.data?.state);
 				setLoaderForLogin(false);
 				if (response instanceof AxiosError) {
 					showError(modifyError(response, {}, { onClearGitHubToken: clearGitHubToken, onRelogin: reLogin }));
 					analyticsClient.sendTrackEvent({ actionSubject: "finishOAuthFlow", action: "fail" });
 					return;
 				} else {
-					analyticsClient.sendTrackEvent({ actionSubject: "finishOAuthFlow", action: "success" });
+					analyticsClient.sendTrackEvent({ actionSubject: "finishOAuthFlow", action: response === true ?  "success" : "fail" });
```

Comment:
```suggestion
					analyticsClient.sendTrackEvent({ actionSubject: "finishOAuthFlow", action: response ?  "success" : "fail" });
```

Reply:
Nah, this is intentional, I know ts checked it, but just to make sure we are only sending "success" when the response is exactly "true"

Reply:
Coz otherwise, if the `resposne` is an `string: "something wrong"`, the TS may failed to checked that since the inner code does
```
return e as AxiosError;
```
In that case, `response ?  "success" : "fail"` will return `success`